sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -28,30 +28,44 @@ impl BearerTokenCredentialPolicy {
         Self {
             credential,
             scopes: scopes.into_iter().map(|s| s.into()).collect(),
-            access_token: Arc::new(Mutex::new("UNSET".to_string())), // Need a better default here or use Option
-            last_refresh_time: Arc::new(Mutex::new(0)), // Need a better default here aswell
+            access_token: Arc::new(Mutex::new(None)),
+            last_refresh_time: Arc::new(Mutex::new(None)),
         }
     }
 
-    fn update_data(&self, new_access_token: &str) {
-        let mut my_access_token = self.access_token.lock().unwrap();
-        *my_access_token = new_access_token.to_string();
-        let mut my_last_refresh_time = self.last_refresh_time.lock().unwrap();
-        *my_last_refresh_time = OffsetDateTime::now_utc().unix_timestamp();
+    fn scopes(&self) -> Vec<&str> {
+        self.scopes
+            .iter()
+            .map(String::as_str)
+            .collect::<Vec<&str>>()
+    }
+
+    fn update_token(&self, new_access_token: &AccessToken) {
+        let mut access_token = self.access_token.lock().unwrap();
```

Comment:
Never `unwrap()` in production. It'll crash the app. Functions that can fail and when that failure should bubble up to the caller should return a `Result<T, E>`. We wrap all ours in `azure_core::Result<T>`, for which you should use an appropriate `azure_core::Error` variant or `Other`.